fsspec/implementations/webhdfs.py

```diff
@@ -102,8 +108,14 @@ def __init__(
                     " token"
                 )
             self.pars["delegation"] = token
-        if user is not None:
-            self.pars["user.name"] = user
```

Comment:
If we have user but not password, this should still be set - I don't see that `self.user` gets used at all except for BasicAuth, whereas if it's in `self.pars` as before, it gets added as an HTTP paraneter.

Reply:
Will modify this indeed, but only for the case that user is provided and password not, because otherwise it would add the user.name as argument in the url.

Reply:
:+1:

Reply:
Addressed in [779e520](https://github.com/fsspec/filesystem_spec/pull/1409/commits/779e5207ef3d3fca95503ea9dd58be0932187d49)